src/components/ebay-area-chart/component.js

```diff
@@ -0,0 +1,331 @@
+                            onTick: d.onTick, // sets the onTick flag to keep track of the points enabled status for mouse events
+                            marker: {
+                                enabled: true, // set marker enabled
+                                radius: pointSize, // set the size of marker
+                                lineColor: backgroundColor, // set border color of hover markers
+                                lineWidth: 4, // set border width of hover markers
+                                fillColor: '#000000', // set fill color of markers
+                            },
+                        },
+                        false, // disable auto redraw
+                        false // disable auto animation
+                    );
+                }
+            });
+        });
+        this.chartRef.redraw(); // trigger redraw after all points have been updated
+    }
+    handleMouseOver(e) {
+        // this function is debounced to improve performance
+        this.chartRef.series.forEach((s) => {
+            s.data.forEach((d) => {
+                if (d.x === e.target.x) {
+                    // if active xAxis hover position matches the data point x update the marker to display
+                    d.update(
+                        {
+                            className: 'ebay-area-chart__marker--visible', // sets the classname
+                            onTick: d.onTick, // sets the onTick flag to keep track of the points enabled status for mouse events
+                            marker: {
+                                enabled: true, // set marker enabled
+                                radius: pointSize, // set the size of marker
+                                lineColor: backgroundColor, // set border color of hover markers
+                                lineWidth: 4, // set border width of hover markers
+                                fillColor: '#000000', // set fill color of markers
+                            },
+                        },
+                        false, // disable auto redraw
+                        false // disable auto animation
+                    );
+                } else if (!d.onTick && d.className !== null) {
+                    d.update(
+                        {
+                            className: null, // nullify className if not active
+                            onTick: d.onTick, // sets the onTick flag to keep track of the points enabled status for mouse events
+                            marker: {
+                                enabled: false, // disable marker
+                            },
+                        },
+                        false, // disable auto redraw
+                        false // disable auto animation
+                    );
+                }
+            });
+        });
+        this.chartRef.redraw(); // trigger redraw after all points have been updated
+    }
+}
```

Comment:
Do we need any `teardown` methods?

Reply:
not for redraw, redraw just tells highcharts to run it's basic draw method which Im suppressing above in the .update calls by the second and third parameters that are set to false.

Reply:
No, not for redraw. I'm pointing at the end of the file
Isn't there a method used to tear down any event listeners so we don't have memory leaks?

Reply:
I could add something